### PR TITLE
Color: Expose OKHSL properties

### LINKED ISSUE #7954: Create public to_ok_hsl functions
### Describe the project you are working on

A 2D world map game with a variety of different countries in different colours.

### Describe the problem or limitation you are having in your project

Not currently able to create hsl colours from the base `Color` function.

I want to be able to create different variations of a rgb colour, through changing the lightness values, i.e. having light red troops for the red territory. This new function would allow it to be programmatically generated. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Creating a new `Color` function to allows `rgb` to be converted into `ok_hsl`.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

This would just involve exporting an already existing `srgb_to_okhsl` function:

https://github.com/godotengine/godot/blob/0ca8542329888e8dccba89d59d3b728090c29991/thirdparty/misc/ok_color.h#L542-L588

It could work like this:
```gdscript
var red = Color.RED
red_hsl = red.to_ok_hsl()
red_hsl.l += 10 # Makes red lighter by adding 10 to its lightness value
var light_red = Color.from_ok_hsl(red)
```

It would also involve creating a new `ok_hsl` class for easily passing around hsl values.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, it would involve having to rewrite all this code:
https://bottosson.github.io/posts/colorpicker/

### Is there a reason why this should be core and not an add-on in the asset library?

This is already a standard library feature, which is currently not exported and it would be a simple implementation.

### PR DESCRIPTION
**Pull request by Hysterelius**
I’m proposing a solution to enhance the `Color` class by adding an `OkHSL` lightness attribute. This attribute will facilitate the creation of colour palettes based on predefined colour stops. Currently, this PR exposes a previously private function to public gdscript. I’m unsure how to extend this functionality to C#, and would appreciate any help on this matter.

This PR is an implementation of my [feature proposal](https://github.com/godotengine/godot-proposals/issues/7954), just aiming to expose pre-existing infrastructure to the user.

I’ve tested these changes locally and everything appears to be working correctly, but I welcome any feedback or suggestions on my PR’s implementation.

_Production edit: Closes https://github.com/godotengine/godot-proposals/issues/7954_

### CODE REVIEW

**Comment by AThousandShips** [comment]
You haven't exposed the `ok_hsl_h/s` values, they are not the same as the `h/s` values

File: doc/classes/Color.xml
Comment (lostminds) [inline doc/classes/Color.xml:508]: Shouldn't this be a float like the others?

**Comment by fire** [comment]
Ok with the proposal but the cicd has to pass and colours are 32 bit floats.

**Comment by Hysterelius** [comment]
> colours are 32 bit floats.

Yeah, probably should have realised that...

> You haven't exposed the ok_hsl_h/s values, they are not the same as the h/s values

I feel like as Godot already has similar (but not the same) `h`/`s` values, it is fine just to add the `ok_hsl_l` value as Godot currently has no value like it. _But this is my personal opinion, and it would very easy to add._

**Comment by AThousandShips** [comment]
What use would it be to *only* have the l value and not the h or s values?

**Comment by Hysterelius** [comment]
I have now added all the values for `ok_hsl` (`s` and `h`) and rebuilt Godot to fix the docs.

**Comment by Hysterelius** [comment]
Can this sneak into the 4.2 release @lostminds?

**Comment by lostminds** [comment]
> Can this sneak into the 4.2 release @lostminds?

@Hysterelius I have no say or insight regarding that, I'm just a member of the community. Merging of pull requests and which branch they will go into is decided on by the various Godot core contributor teams. You can see below that this particular one will need review and approval by the core and documentation teams.

**Comment by Hysterelius** [comment]
Sorry, I just assumed you would have to finish your review before the other teams could look at it.

**Review by lostminds — APPROVED** [review]

**Comment by AThousandShips** [comment]
4.2 is in feature freeze so will likely have to wait for 4.3, this PR was opened just around the time of feature freeze

**Comment by nonchip** [comment]
@AThousandShips is this really a feature though? feels more like a fix for forgetting to expose the already existing feature in a usable manner...

**Comment by AThousandShips** [comment]
That falls under features, it's not a bug but missing functionality, in either case we are close to release so it can soon be looked into once the 4.3 cycle starts

**Comment by Kermalis** [comment]
Bumping this. I am using Color.FromOkHsl in C# but have no method to convert back to 3 [0,1] values.

**Comment by AThousandShips** [comment]
@Kermalis Please don't bump without contributing significant new information. Use the :+1: reaction button on the first post instead.

**Review by fire — APPROVED** [review]
Looks good to me.

I needed a way to reduce the lightness of an existing red colour.

The theory is I'd from a red colour I'd get the existing okhsl values from this enhancement, then use to half the lightness value. `from_ok_hsl(h: float, s: float, l: float, alpha: float = 1.0)`

Sorry for the delay!

**Review by AThousandShips — APPROVED** [review]

**Comment by AThousandShips** [comment]
Please squash your commits into one, see [the interactive rebase](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

**Comment by akien-mga** [comment]
Amended the commit message to be clearer, and tweaked the spelling of OKHSL (it was "OkHSL" in the docs).
Should be good to merge.

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**
